### PR TITLE
fix: correct order of amount in and out payload args

### PR DESCRIPTION
**Pull request by wjw12**
Aside:
- `encodeRoute` accepts an optional parameter `balanceCoinIn` to handle the case where user balance may be smaller than expected input amount + slippage
- configure eslint correctly
- update router deployment address

### CODE REVIEW

File: src/ThalaswapRouter.ts [hunk 1]
```diff
@@ -211,7 +211,14 @@ class ThalaswapRouter {
     );
   }
 
-  encodeRoute(route: Route, slippagePercentage: number): EntryPayload {
+  // balanceCoinIn is the user's balance of input coin. If it's specified, this function will check
+  // (1) for exact-in type of swap, throw an error if the user has enough balance to perform the swap.
```
Comment (0xbe1) [inline src/ThalaswapRouter.ts:215]: if the user has -> if the user doesn't have

File: src/ThalaswapRouter.ts [hunk 2]
```diff
@@ -222,22 +229,30 @@ class ThalaswapRouter {
...
-              tokenInDecimals,
-            ),
-          ];
+    let amountInArg: number;
+    let amountOutArg: number;
+    if (route.type === "exact_input") {
+      if (balanceCoinIn !== undefined && balanceCoinIn < route.amountIn) {
+        throw new Error("Insufficient balance");
+      }
+      (amountInArg = scaleUp(route.amountIn, tokenInDecimals)),
```
Comment (0xbe1) [inline src/ThalaswapRouter.ts:238]: the bracket looks a bit odd 🤔

**Review by 0xbe1 — APPROVED** [review]
lgtm

**Review by SamuelQZQ — APPROVED** [review]
lgtm

File: src/ThalaswapRouter.ts (hunk 2 above)
Comment (wjw12) [reply src/ThalaswapRouter.ts:238]: removed

**Pull request merged**